### PR TITLE
Updated docs for PG 17 support

### PR DESCRIPTION
**Pull request by nastena1606**

### CODE REVIEW

File: docs/reference.md [hunk 1]
```diff
@@ -62,8 +62,10 @@ The following table describes the `pg_stat_monitor` view for PostgreSQL 16 and h
      | local_blks_written | bigint                      | Total number of local blocks  written simultaneously to a drive during the query execution
      | temp_blks_read     | bigint                      | Total number of blocks of temporary files read from a drive. Temporary files are used when there's not enough memory to execute a query
      | temp_blks_written  | bigint                      | Total number of blocks of temporary files written to a drive
-     | blk_read_time      | double precision            | Total waiting time (in ms) for reading blocks
-     | blk_write_time     | double precision            | Total waiting time (in ms) for writing blocks to a drive
+     | shared_blk_read_time      | double precision            | Total waiting time (in ms) for reading shared blocks
+     | shared_blk_write_time     | double precision            | Total waiting time (in ms) for writing shared blocks to a drive
+     | local_blk_read_time       | double precision  | Total time the statement spent on reading local blocks, in ms
+     | local_blk_write_time      | double precision  | Total time the statement spent on writing local blocks, in ms
```
Comment (artemgavrilov) [inline docs/reference.md:68]:
```suggestion
     | shared_blk_read_time      | double precision            | Total time the statement spent reading shared blocks, in milliseconds
     | shared_blk_write_time     | double precision            | Total time the statement spent writing shared blocks, in milliseconds
     | local_blk_read_time       | double precision  | Total time the statement spent on reading local blocks, in milliseconds
     | local_blk_write_time      | double precision  | Total time the statement spent on writing local blocks, in milliseconds
```

Let's use descriptions from `pg_stat_statements` docs

File: docs/reference.md [hunk 2]
```diff
@@ -62,8 +62,10 @@ The following table describes the `pg_stat_monitor` view for PostgreSQL 16 and h
...
      | temp_blk_read_time | double precision            | Total number of temp blocks read by the statement 
      | temp_blk_write_time| double precision            | Total number of temp blocks written by the statement
```
Comment (artemgavrilov) [inline docs/reference.md:70]: Fix descriptions
```suggestion
     | temp_blk_read_time | double precision            | Total time the statement spent reading temporary file blocks, in milliseconds 
     | temp_blk_write_time| double precision            | Total time the statement spent writing temporary file blocks, in milliseconds
```

File: docs/reference.md [hunk 3]
```diff
@@ -80,8 +82,82 @@ The following table describes the `pg_stat_monitor` view for PostgreSQL 16 and h
...
+     | local_blks_dirtied | bigint                      | Total number of local blocks "dirtied" during the query execution (i.e. a query modified at least one tuple in a block and this block must be written to a drive)
+     | local_blks_written | bigint                      | Total number of local blocks  written simultaneously to a drive during the query execution
+     | temp_blks_read     | bigint                      | Total number of blocks of temporary files read from a drive. Temporary files are used when there's not enough memory to execute a query
+     | temp_blks_written  | bigint                      | Total number of blocks of temporary files written to a drive
+     | blk_read_time      | double precision            | Total waiting time (in ms) for reading blocks
+     | blk_write_time     | double precision            | Total waiting time (in ms) for writing blocks to a drive
+     | temp_blk_read_time | double precision            | Total number of temp blocks read by the statement 
+     | temp_blk_write_time| double precision            | Total number of temp blocks written by the statement
```
Comment (artemgavrilov) [inline docs/reference.md:146]:
```suggestion
     | temp_blk_read_time | double precision            | Total time the statement spent reading temporary file blocks, in milliseconds
     | temp_blk_write_time| double precision            | Total time the statement spent writing temporary file blocks, in milliseconds
```

File: docs/reference.md [hunk 4]
```diff
@@ -80,8 +82,82 @@ The following table describes the `pg_stat_monitor` view for PostgreSQL 16 and h
...
+     | local_blks_hit     | bigint                      | The number of blocks which are considered as local by the backend and thus are used for temporary tables
+     | local_blks_read    | bigint                      | Total number of local blocks read during the query execution
+     | local_blks_dirtied | bigint                      | Total number of local blocks "dirtied" during the query execution (i.e. a query modified at least one tuple in a block and this block must be written to a drive)
+     | local_blks_written | bigint                      | Total number of local blocks  written simultaneously to a drive during the query execution
+     | temp_blks_read     | bigint                      | Total number of blocks of temporary files read from a drive. Temporary files are used when there's not enough memory to execute a query
+     | temp_blks_written  | bigint                      | Total number of blocks of temporary files written to a drive
+     | blk_read_time      | double precision            | Total waiting time (in ms) for reading blocks
+     | blk_write_time     | double precision            | Total waiting time (in ms) for writing blocks to a drive
```
Comment (artemgavrilov) [inline docs/reference.md:144]:
```suggestion
     | blk_read_time      | double precision            | Total time the statement spent reading data file blocks, in milliseconds
     | blk_write_time     | double precision            | Total time the statement spent writing data file blocks, in milliseconds
```

File: docs/reference.md [hunk 5]
```diff
@@ -80,8 +82,82 @@ The following table describes the `pg_stat_monitor` view for PostgreSQL 16 and h
      | jit_optimization_time| double precision     | Total time spent by the statement on optimizing, in milliseconds 
      | jit_emission_count   | bigint               | Number of times code has been emitted
      | jit_emission_time    | double precision     | Total time spent by the statement on emitting code, in milliseconds
+     | jit_deform_count     | bigint               | Total number of tuple deform functions JIT-compiled by the statement | 
+     jit_deform_time       | double precision | Total time spent by the statement on JIT-compiling tuple deform functions, in milliseconds | 
+     | stats_since         | timestamp with time zone| Time at which statistics gathering started for this statement|
+     | minmax_stats_since  | timestamp with time zone| Time at which min/max statistics gathering started for this statement (fields min_plan_time, max_plan_time, min_exec_time and max_exec_time) |
```
Comment (artemgavrilov) [inline docs/reference.md:88]:
```suggestion
     | minmax_stats_since  | timestamp with time zone| Time at which min/max statistics gathering started for this statement (fields min_plan_time, max_plan_time, min_exec_time and max_exec_time). Please note that unlike `pg_stat_statements` `pg_stat_monitor` doesn't have option to reset min/max counters, so `minmax_stats_since` always equals to `stats_since` |
```

**Review by artemgavrilov — APPROVED** [review]

**Comment by artemgavrilov** [comment]
@nastena1606 We merged code related to PG17, so I guess we can merge documentation.

**Comment by nastena1606** [comment]
I would keep it unmerged for now until we release PPG 17. Plus should other changes arise, I can add them within this request. @janwieremjewicz what's your thought on this?

**Pull request merged**